plugins/nanospell/plugin.js

```diff
@@ -590,24 +594,37 @@
 			 for a given range, get the unique words in it that we don't have a spellcheck status for
 			 */
 			function scanWordsInRange(range) {
-				var block;
-				var iterator = range.createIterator();
+				var combinedText = '',
+					block,
+					blockList = [],
+					iterator = range.createIterator();
 				while (( block = iterator.getNextParagraph() )) {
 					block.setCustomData('spellCheckInProgress', true);
-					var unknownWords = getUnknownWords(block.getText());
-					startCheckOrMarkWords(unknownWords, block);
+					combinedText += block.getText() + ' ';
+					blockList.push(block);
+					if (blockList.length === BLOCK_REQUEST_LIMIT) {
+						startCheckOrMarkWords(getUnknownWords(combinedText), blockList);
+						combinedText = '';
+						blockList = [];
+					}
+				}
+				if (blockList.length > 0) {
+					startCheckOrMarkWords(getUnknownWords(combinedText), blockList);
```

Comment:
Yeah, this calls the next step for any remaining blocks when the number of blocks is not perfectly divisible by whatever the `BLOCK_REQUEST_LIMIT` is set to... which is most of the time